account_move_base_import/models/account_journal.py

```diff
@@ -0,0 +1,384 @@
+        :param calls: list of lookup function name available in rules
+        :param dict line: read of the concerned account.bank.statement.line
+        :return:
+            A dict of value that can be passed directly to the write method of
+            the statement line or {}
+           {'partner_id': value,
+            'account_id: value,
+            ...}
+        """
+        if not calls:
+            calls = self._get_rules()
+        rule_obj = self.env['account.move.completion.rule']
+        for call in calls:
+            method_to_call = getattr(rule_obj, call.function_to_call)
+            result = method_to_call(line)
+            if result:
+                result['already_completed'] = True
+                return result
+        return None
+
+    @api.multi
+    def _prepare_counterpart_line(self, move, amount, date):
+        if amount > 0.0:
+            account_id = self.default_debit_account_id.id
+            credit = 0.0
+            debit = amount
+        else:
+            account_id = self.default_credit_account_id.id
+            credit = -amount
+            debit = 0.0
+        counterpart_values = {
+            'name': _('/'),
+            'date_maturity': date,
+            'credit': credit,
+            'debit': debit,
+            'partner_id': self.partner_id.id,
+            'move_id': move.id,
+            'account_id': account_id,
+            'already_completed': True,
+            'journal_id': self.id,
+            'company_id': self.company_id.id,
+            'currency_id': self.currency_id.id,
+            'company_currency_id': self.company_id.currency_id.id,
+            'amount_residual': amount,
+        }
+        return counterpart_values
+
+    @api.multi
+    def _create_counterpart(self, parser, move):
+        move_line_obj = self.env['account.move.line']
+        refund = 0.0
+        payment = 0.0
+        transfer_lines = []
+        for move_line in move.line_ids:
+            refund -= move_line.debit
+            payment += move_line.credit
+        if self.split_counterpart:
+            if refund:
+                transfer_lines.append(refund)
+            if payment:
+                transfer_lines.append(payment)
+        else:
+            total_amount = refund + payment
+            if total_amount:
+                transfer_lines.append(total_amount)
+        counterpart_date = parser.get_move_vals().get('date') or \
+            fields.Date.today()
+        transfer_line_count = len(transfer_lines)
+        check_move_validity = False
+        for amount in transfer_lines:
+            transfer_line_count -= 1
+            if not transfer_line_count:
+                check_move_validity = True
+            vals = self._prepare_counterpart_line(move, amount,
+                                                  counterpart_date)
+            move_line_obj.with_context(
+                check_move_validity=check_move_validity
```

Comment:
seems  check_move_validity=True will just execute _post_validate on lines, so we can perform this explicitly  and rid of transfer_line_count

Reply:
I'm not sure to understand what you mean here, but I'd prefer not to change what has been working in previous versions.

Reply:
if I read this correctly lines 120-130 can be replaced: 

        for amount in transfer_lines:
            vals = self._prepare_counterpart_line(move, amount,
                                                  counterpart_date)
            lines += move_line_obj.with_context(
                check_move_validity=False
            ).create(vals)
         move._post_validate()

Reply:
`check_move_validate` is only used on the last iteration (i.e last item to be created from `transfer_lines`, so this wouldn't do exactly the same... :man_shrugging: